evennia/contrib/currency.py

```diff
@@ -0,0 +1,374 @@
+    def __init__(self, msg):
+        self.msg = msg
+
+class CurrencyHandler(object):
+    def __init__(self, obj, db_attribute='currency'):
+        if not obj.attributes.has(db_attribute):
+            obj.attributes.add(db_attribute, {})
+
+        self.attr_dict = obj.attributes.get(db_attribute)
+        self.cache = {}
+
+    def __len__(self):
+        # Return the number of values in 'attr_dict'.
+        return len(self.attr_dict)
+
+    def __setattr__(self, key, value):
+        # Return an error message if attrib is assigned directly
+        if key in ('attr_dict', 'cache'):
+            super(CurrencyHandler, self).__setattr__(key, value)
+        else:
+            raise CurrencyException(
+                "Currency object not settable. Assign one of " + 
+                "the monetary conversion properties instead."
+            )
+
+    def __setitem__(self, key, value):
+        # Returns error message if currency objects are assigned directly.
+        return self.__setattr__(key, value)
+
+    def __getattr__(self, currency):
+        # Returns currency instances accessed as attributes.
+        return self.get(currency)
+
+    def __getitem__(self, currency):
+        # Returns 'Currency' instances accessed as dict keys.
+        return self.get(currency)
+
+    def get(self, currency):
+        """
+        Args: 
+            currency (str): key from the currency dict containing config data
+            for the currency type. "all" returns a list of all currency keys
+        Returns:
+            ('Currency' or 'None'): named Currency class or None if currency key
+            is not found in currency collection.
+        """
+        if currency not in self.cache:
+            if currency not in self.attr_dict:
+                return None
+            data = self.attr_dict[currency]
+            self.cache[currency] = Currency(data)
+        return self.cache[currency]
+
+    def add(self, key, name, value, amt):
+        # Create a new Currency and add it to the handler.
+        if key in self.attr_dict:
+            raise CurrencyException("Currency '{}' already exists.".format(key))
+
+        currency = dict(name=name,
+                     value=value, 
+                     amt=amt)
+
+        self.attr_dict[key] = currency
+
+    def remove(self, currency):
+        # Remove a Currency type from the handler's parent object.
+        if currency not in self.attr_dict:
+            raise CurrencyException("Currency not found: {}".format(currency))
+
+        if currency in self.cache:
+            del self.cache[currency]
+        del self.attr_dict[currency]
+
+    def clear(self):
+        # Remove all Currency from the handler's parent object.
+        for currency in self.all:
+            self.remove(currency)
+
+    @property
+    def all(self):
+        # Return a list of all currency in this CurrencyHandler.
+        return self.attr_dict.keys()
+
+    @property
+    def contents(self):

```

Comment:
While the name `contents` fits well for your `purse` analogy, I don't think it fits as a generic handler name. This is a pretty-print method. To be consistent with other similar methods in Evennia, something like `to_string` is a better name.